ambry-frontend/src/main/java/com.github.ambry.frontend/AmbryUrlSigningService.java

```diff
@@ -108,20 +122,35 @@ public String getSignedUrl(RestRequest restRequest) throws RestServiceException
           case RestUtils.Headers.MAX_UPLOAD_SIZE:
             maxUploadSize = RestUtils.getLongHeader(args, RestUtils.Headers.MAX_UPLOAD_SIZE, true);
             break;
+          case RestUtils.Headers.CHUNK_UPLOAD:
+            chunkUpload = RestUtils.getBooleanHeader(args, RestUtils.Headers.CHUNK_UPLOAD, true);
```

Comment:
This is because intellij has some return analysis and sees that there is a case where this could return a null object (if `required` is false). However, it is not smart enough to know that when `required` is true this function will throw an exception instead of returning null.